src/LtiServiceConnector.php

```diff
@@ -172,6 +194,47 @@ public function getAll(
         return $results;
     }
 
+    public function logRequest(
+        int $requestType,
+        IServiceRequest $request,
+        array $responseHeaders,
+        ?array $responseBody
+    ): void {
+        $contextArray = [
+            'request_method' => $request->getMethod(),
+            'request_url' => $request->getUrl(),
+            'response_headers' => $responseHeaders,
+            'response_body' => json_encode($responseBody),
+        ];
+        $userId = '';
+
+        $payload = $request->getPayload();
+        if (isset($payload['body'])) {
+            $requestBody = $payload['body'];
+
+            $contextArray['request_body'] = $requestBody;
+
+            if (isset(json_decode($requestBody)->userId)) {
+                $userId = json_decode($requestBody)->userId;
+            }
+        }
```

Comment:
`$userId` isn't in the requests for creating or getting lineitems, currently it only gets set when syncing a grade.

Reply:
Do you think we could simplify all this `if` logic with the [null coalescing assignment](https://stitcher.io/blog/shorthand-comparisons-in-php#null-coalescing-assignment-operator)? E.g. `$request->getPayload()['body'])->userId ??= ''`, meaning that if that value exists, it will be used, otherwise, it's just an empty string